Modules/Messages/src/Messages.Application/Messages/Queries/GetMessage/Handler.cs

```diff
@@ -1,38 +1,30 @@
 ﻿using AutoMapper;
-using Backbone.Modules.Messages.Application.Extensions;
-using Backbone.Modules.Messages.Application.Infrastructure.Persistence;
+using Backbone.Modules.Messages.Application.Infrastructure.Persistence.Repository;
 using Backbone.Modules.Messages.Application.Messages.DTOs;
-using Backbone.Modules.Messages.Domain.Entities;
 using Enmeshed.BuildingBlocks.Application.Abstractions.Infrastructure.UserContext;
 using MediatR;
 
 namespace Backbone.Modules.Messages.Application.Messages.Queries.GetMessage;
 
 public class Handler : IRequestHandler<GetMessageQuery, MessageDTO>
 {
-    private readonly IMessagesDbContext _dbContext;
+    private readonly IMessagesRepository _messagesRepository;
     private readonly IMapper _mapper;
     private readonly MessageService _messageService;
     private readonly IUserContext _userContext;
 
-    public Handler(IMessagesDbContext dbContext, IUserContext userContext, IMapper mapper, MessageService messageService)
+    public Handler(IUserContext userContext, IMapper mapper, MessageService messageService, IMessagesRepository messagesRepository)
     {
-        _dbContext = dbContext;
         _userContext = userContext;
         _mapper = mapper;
         _messageService = messageService;
+        _messagesRepository = messagesRepository;
     }
 
     public async Task<MessageDTO> Handle(GetMessageQuery request, CancellationToken cancellationToken)
     {
-        var addressOfActiveIdentity = _userContext.GetAddress();
-
-        var message = await _dbContext
-            .Set<Message>()
-            .IncludeAllReferences(addressOfActiveIdentity)
-            .WithSenderOrRecipient(_userContext.GetAddress())
-            .FirstWithId(request.Id, cancellationToken);
-
+        var message = await _messagesRepository.Find(request.Id, _userContext.GetAddress(), cancellationToken);
```

Comment:
Here a reminder of the comment in my last review, because I think that this is still a problem:

> You didn't test your changes, did you? Because I think there will be an error on the GET /Messages and the GET /Messages/{id} routes. You forgot to remove the IMessagesDbContext interface. If you delete it, you will find another usage, which is in a class called MessageService, with the two methods MarkMessageAsReceived and MarkMessagesAsReceived. These two methods update the passed Messages and save them in the database. The methods are called from the query handlers for GetMessageQuery and ListMessagesQuery. And since these handlers now use the repository, which uses the _readOnlyDbSet to read them, they can't be saved, because they are untracked (_readOnlyMessages = dbContext.Messages.AsNoTracking()).

Reply:
Using the debugger and checking on the database directly, I have confirmed that the way the message is fetched does not impact the write; possibly because it takes place with the appropriate dbContext, via the repo.
i.e., the expected behaviour is working.

Reply:
Okay, maybe they changed this behavior. But I think we should still use the tracked queries here. Because 
1. it is recommended in the docs: https://learn.microsoft.com/en-us/ef/core/querying/tracking
2. if the Messages are untracked, I would bet that even though you only update a single property, the whole object is updated on the database, resulting in an unnecessary amount of data sent to the database.

I think we should add a parameter `track` of type `bool` that defaults to `false`, and that we set to `true` in case we want to edit the Message(s)